pkg/reconciliation/reconcile_datacenter.go

```diff
@@ -27,6 +30,59 @@ func (rc *ReconciliationContext) ProcessDeletion() result.ReconcileResult {
 		return result.Error(err)
 	}
 
+	origSize := rc.Datacenter.Spec.Size
+	if rc.Datacenter.Status.GetConditionStatus(api.DatacenterDecommission) == corev1.ConditionTrue {
+		rc.Datacenter.Spec.Size = 0
```

Comment:
How come you set `.Spec.Size` here? Is it necessary?

Reply:
This code is called before actual decommission:

https://github.com/burmanm/cass-operator/blob/decommission_dc/pkg/reconciliation/reconcile_racks.go#L51

So setting it here ensures that the CalculateRackInformation has Size set to 0